python/pyspark/pandas/tests/data_type_ops/test_boolean_ops.py

```diff
@@ -572,10 +578,28 @@ def test_isnull(self):
     def test_astype(self):
         pser = self.pser
         psser = self.psser
+
+        # [True, False, <NA>] is returned in pandas
         self.assert_eq(["True", "False", "None"], self.psser.astype(str).tolist())
+
         self.assert_eq(pser.astype("category"), psser.astype("category"))
         cat_type = CategoricalDtype(categories=[False, True])
         self.assert_eq(pser.astype(cat_type), psser.astype(cat_type))
+        for dtype in self.extension_dtypes:
+            if extension_float_dtypes_available:
+                is_extension_float_dtype = dtype in [
+                    "Float32",
+                    "Float64",
+                    Float32Dtype(),
+                    Float64Dtype(),
+                ]
+
+                if is_extension_float_dtype:
+                    self.assert_eq([1.0, 0.0, np.nan], self.psser.astype(dtype).tolist())
```

Comment:
Could you add a comment why we need this?

Reply:
Sure added.